app/models/user.rb

```diff
@@ -7,9 +7,32 @@ class User < ApplicationRecord
 
   has_one :shop, dependent: :destroy
   has_one_attached :avatar
+  has_many :cart_items, dependent: :destroy
 
   after_create :attach_avatar
 
+  def cart_items_group_by_shop 
+    return nil if cart_items.blank? 
+
+    result = [] 
+
+    cart_items.group_by(&:shop).each { |shop, items| result.push({ shop: shop, items: items })  }
+
+    result
```

Comment:
this could be cleaner by
```
[].tap do |result|
  cart_items.group_by(&:shop).each { |shop, items| result.push({ shop: shop, items: items })  }
end
```

Reply:
https://stackoverflow.com/questions/17493080/advantage-of-tap-method-in-ruby

Reply:
noted and updated